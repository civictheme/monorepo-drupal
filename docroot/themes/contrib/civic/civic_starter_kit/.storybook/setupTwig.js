const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
