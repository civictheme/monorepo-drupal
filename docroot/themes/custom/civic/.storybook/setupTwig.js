const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
