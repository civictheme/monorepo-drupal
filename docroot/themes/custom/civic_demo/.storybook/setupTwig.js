const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
