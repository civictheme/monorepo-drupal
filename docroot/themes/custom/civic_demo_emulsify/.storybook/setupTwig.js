const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
