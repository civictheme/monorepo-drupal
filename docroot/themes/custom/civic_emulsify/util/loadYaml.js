import { resolve } from 'path';
import { readFileSync } from 'fs';
import { parse } from 'yaml';
import R from 'ramda';

/**
 * Small utility function that loads a yaml file and parses it synchronously.
 * This is intended to make composition cleaner.
 *
 * @param {string} relativePath - relative path to a yaml file that will be loaded and parsed.
 *
 * @returns {string} JavaScript object that results from the yaml parsing of the specified file.
 */
export default function loadYaml(relativePath) {
  const fullPath = resolve(__dirname, relativePath);
  return parse(readFileSync(fullPath, 'utf8'));
}
