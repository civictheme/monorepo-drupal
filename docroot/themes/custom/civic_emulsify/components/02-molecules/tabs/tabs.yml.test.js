import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import loadYaml from '../../../util/loadYaml';
import { setupTwig, namespaces } from '../../../.storybook/setupTwig';

setupTwig(Twig);

// @TODO: the javascript file associated with this component
// needs to be tested.
describe('tabs', () => {
  it('can render tabs', async () => {
    const { container } = await render(
      join(__dirname, 'tabs.twig'),
      loadYaml(join(__dirname, 'tabs.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });
});
