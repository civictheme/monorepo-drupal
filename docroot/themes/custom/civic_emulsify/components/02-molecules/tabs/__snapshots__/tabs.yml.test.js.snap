// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabs can render tabs 1`] = `
<div>
  





  
  <div
    class="tabs no-js"
    id="tabs"
  >
    
    
    <ul
      class="tabs__nav"
    >
      
              



      <li>
        
  
      
        <a
          _keys="class"
          class=" tabs__link--local-tasks"
          href="null"
        >
          null
        </a>
        
  
      </li>
      
              



      <li>
        
  
      
        <a
          _keys="class"
          class=" tabs__link--local-tasks"
          href="null"
        >
          null
        </a>
        
  
      </li>
      
              



      <li>
        
  
      
        <a
          _keys="class"
          class=" tabs__link--local-tasks"
          href="null"
        >
          null
        </a>
        
  
      </li>
      
          
    </ul>
    
          
    <div
      class="tabs__tab is-active"
    >
      
        
      <div
        class="tabs__content"
        id="tab-0"
      >
        
          
        <p>
          Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat, vel illum dolore eu feugiat nulla facilisis at vero eros et accumsan et iusto odio dignissim qui blandit praesent luptatum zzril delenit augue duis dolore te feugait nulla facilisi. Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat.
        </p>
        
        
      </div>
      
      
    </div>
    
          
    <div
      class="tabs__tab"
    >
      
        
      <div
        class="tabs__content"
        id="tab-1"
      >
        
          
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
        </p>
        
        
      </div>
      
      
    </div>
    
          
    <div
      class="tabs__tab"
    >
      
        
      <div
        class="tabs__content"
        id="tab-2"
      >
        
          
        <p>
          Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat, vel illum dolore eu feugiat nulla facilisis at vero eros et accumsan et iusto odio dignissim qui blandit praesent luptatum zzril delenit augue duis dolore te feugait nulla facilisi.
        </p>
        
        
      </div>
      
      
    </div>
    
      
  </div>
  

</div>
`;
